File: DSA PRACTICE/LECTURE10/ReverseArray.java
// Reverse an Array Using Swapping (For Loop)
import java.util.Scanner;

class ReverseArray
{
	public static void reverseArray(int arr[],int size)
	{
		int start, end=size-1, temp=0;

		for(start=0; start<size/2; start++)
		{
			temp=arr[start];
			arr[start]=arr[end-start];
			arr[end-start]=temp;
		}
	}

	public static void printArray(int arr[],int size)
	{
		for(int i=0;i<size;i++)
		{
			System.out.print(arr[i]+" ");
		}
	}

	public static void main(String[] args)
	{
		Scanner sc=new Scanner(System.in);
		System.out.print("Enter the size: ");
		int size=sc.nextInt();

		System.out.println("Input the elements");
		int arr[]=new int[100];

		for(int i=0;i<size;i++)
		{
			arr[i]=sc.nextInt();
		}

		reverseArray(arr,size);
		printArray(arr,size);
	}
}

/*
OUTPUT:
Enter the size: 5
Input the elements
1
2
3
4
5
5 4 3 2 1
OUTPUT:
Enter the size: 4
Input the elements
1
2
3
4
4 3 2 1
*/

File: DSA PRACTICE/LECTURE 20/ReverseArray.cpp
// PROGRAM 01: REVERSE AN ARRAY(Code Studio Problem) 

#include <bits/stdc++.h> 
void reverseArray(vector<int> &arr , int m)
{
        int start=m+1;
        int end=arr.size()-1;
        while(start<=end)
        {         
            swap(arr[start], arr[end]);
            start++;
            end--;
        }  
}


/*
INPUT:
m=3
arr={1 2 3 4 5 6}

OUTPUT:
1 2 3 4 6 5

INPUT:
m=2
arr={10 9 8 7 6}

OUTPUT:
10 9 8 6 7

*/
// PROBLEM: https://bit.ly/3EOyAFz